tests/e2e.bats

```diff
@@ -3,14 +3,68 @@
 load 'test_helper/bats-support/load'
 load 'test_helper/bats-assert/load'
 
+# ---- GLOBAL CLEANUP ----
+
 teardown() {
-  # Cleanup all dummy devices on worker nodes
-  docker exec "$CLUSTER_NAME"-worker bash -c "ip -br link show type dummy | awk '{print \$1}' | xargs -I {} ip link delete {}"
-  docker exec "$CLUSTER_NAME"-worker2 bash -c "ip -br link show type dummy | awk '{print \$1}' | xargs -I {} ip link delete {}"
+  if [[ -z "$BATS_TEST_COMPLETED" || "$BATS_TEST_COMPLETED" -ne 1 ]] && [[ -z "$BATS_TEST_SKIPPED" ]]; then
+    dump_debug_info_on_failure
+  fi
+  cleanup_k8s_resources
+  cleanup_dummy_interfaces
+  cleanup_bpf_programs
+  sleep 2
+}
 
-  sleep 5
+dump_debug_info_on_failure() {
```

Comment:
Not needed on this but I wonder if we can dump these into a file that we can later read?